### PR TITLE
Update deps and CircleCI for OTP 22

### PR DESCRIPTION
**Pull request by uwiger**
See issue [#3293](https://github.com/aeternity/aeternity/issues/3293)

This PR updates deps and CircleCI configuration to OTP 22.3.4.10
While we were at it, the OTP 21 build (kept for now) was updated to 21.3.8.17

### CODE REVIEW

**Comment by gorbak25** [comment]
aehttp_spec_SUITE fails...

File: .circleci/config.yml [hunk 1]
```diff
@@ -106,7 +106,7 @@ references:
         LIBSODIUM_VERSION: "1.0.16"
       # source: https://github.com/aeternity/docker-builder/blob/master/Dockerfile#L23
       command: |
-        LIBSODIUM_DOWNLOAD_URL="https://github.com/jedisct1/libsodium/releases/download/${LIBSODIUM_VERSION}/libsodium-${LIBSODIUM_VERSION}.tar.gz" \
+        LIBSODIUM_DOWNLOAD_URL="https://github.com/jedisct1/libsodium/releases/downloadw/${LIBSODIUM_VERSION}/libsodium-${LIBSODIUM_VERSION}.tar.gz" \
```
Comment (dincho) [inline .circleci/config.yml:109]: this change looks like mistake ?

File: .circleci/config.yml [hunk 2]
```diff
@@ -32,17 +32,17 @@ references:
     resource_class: large
     working_directory: ~/aeternity
     environment:
-      OTP_VERSION: 20.3.8.24
-  container_otp_plt: &container_otp_plt _build/default/rebar3_20.3.8.24_plt
+      OTP_VERSION: 22.3.4.10
```
Comment (dincho) [inline .circleci/config.yml:35]: It seems that erlang repository does not have a package with this version: https://packages.erlang-solutions.com/erlang/debian/pool/

The closest/latest I see is `22.3.4.9`

**Review by dincho — COMMENTED** [review]
I found couple of issues

**Review by dincho — APPROVED** [review]

**Review by gorbak25 — APPROVED** [review]

**Pull request merged**